pkg/apis/internal.admin.acorn.io/v1/baseresources.go

```diff
@@ -75,43 +77,51 @@ func (current *BaseResources) Fits(incoming BaseResources) error {
 		}
 	}
 
-	// Check if any of the quantity resources are exceeded
-	for _, r := range []struct {
-		resource          string
-		current, incoming resource.Quantity
-	}{
-		{"VolumeStorage", current.VolumeStorage, incoming.VolumeStorage},
-		{"Memory", current.Memory, incoming.Memory},
-		{"Cpu", current.CPU, incoming.CPU},
-	} {
-		if !FitsQuantity(r.current, r.incoming) {
-			exceededResources = append(exceededResources, r.resource)
-		}
+	if len(exceededResources) != 0 {
+		errs = append(errs, fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", ")))
 	}
 
-	// Build an aggregated error message for the exceeded resources
-	if len(exceededResources) > 0 {
-		return fmt.Errorf("%w: %s", ErrExceededResources, strings.Join(exceededResources, ", "))
+	if err := current.ComputeClasses.Fits(incoming.ComputeClasses); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := current.VolumeClasses.Fits(incoming.VolumeClasses); err != nil {
+		errs = append(errs, err)
 	}
 
-	return nil
+	// Build an aggregated error message for the exceeded resources
+	return errors.Join(errs...)
 }
 
 // ToString will return a string representation of the BaseResources within the struct.
 func (current *BaseResources) ToString() string {
-	return ResourcesToString(
+	// make sure that an empty string doesn't have a comma
+	result := CountResourcesToString(
 		map[string]int{
 			"Apps":       current.Apps,
 			"Containers": current.Containers,
 			"Jobs":       current.Jobs,
 			"Volumes":    current.Volumes,
 			"Images":     current.Images,
 		},
-		map[string]resource.Quantity{
-			"VolumeStorage": current.VolumeStorage,
-			"Memory":        current.Memory,
-			"Cpu":           current.CPU,
-		})
+	)
+
+	for _, resource := range []struct {
+		name     string
+		asString string
+	}{
+		{"ComputeClasses", current.ComputeClasses.ToString()},
+		{"VolumeClasses", current.VolumeClasses.ToString()},
+	} {
+		if result != "" && resource.asString != "" {
+			result += ", "
+		}
+		if resource.asString != "" {
+			result += fmt.Sprintf("%s: %s", resource.name, resource.asString)
+		}
```

Comment:
nit: I think you could achieve the same thing with `strings.Join`, right?

Reply:
`strings.Join()` will still add the separator if the string is empty which is not what I want. Is there another method that would work better here?